src/MergeContext.js

```diff
@@ -954,36 +964,77 @@ class PullRequest {
         return false;
     }
 
+    async _getMergeCommit() {
+        const mergeSha = await GH.getReference("pull/" + this._prNumber() + "/merge");
+        return await GH.getCommit(mergeSha);
+    }
+
+    // whether we should prevent the failed staged PR (that lost its staging to another PR)
+    // from restaging and failing again, creating a live lock with that other PR
+    async _recalculateStagingFailed() {
+        assert(!this._stagedSha());
+        assert(this._stagingFailed);
+
+        const allEvents = await GH.getIssueEvents(this._prNumber());
+        // we consider all commits created by the bot user and referencing this PR as staged commits
+        let stagedEvents = allEvents.filter(ev => ev.event === "referenced" && ev.actor.login === Config.githubUserLogin());
+        if (!stagedEvents.length)
+            return false;
+
+        // just in case: events should be already sorted by date
+        stagedEvents = stagedEvents.sort((ev1, ev2) => Date.parse(ev1.created_at) - Date.parse(ev2.created_at));
+        const lastStaged = stagedEvents[stagedEvents.length - 1];
+        const mergeCommit = await this._getMergeCommit();
+        const mergeCommitCreatedAt = Date.parse(mergeCommit.author.date);
+        const stagedCommitCreatedAt = Date.parse(lastStaged.created_at);
+        assert(mergeCommitCreatedAt);
+        // whether the merge commit is fresher than the last staged commit
+        if (stagedCommitCreatedAt < mergeCommitCreatedAt)
+            return false;
+
+        const stagedStatuses = await this._getStagingStatuses(lastStaged.commit_id);
+        // TODO: If something made this (previously failed) commit succeed, then
+        // we should use it further, if possible.
+        return stagedStatuses.failed();
+    }
+
     async _loadPrState() {
         if (!this._stagedSha()) {
-            if (await this._mergedSomeTimeAgo())
+            if (await this._mergedSomeTimeAgo()) {
+                // leave this._stagingFailed intact here to
+                // keep labeling of manually merged PRs with failed commits
                 this._enterMerged();
-            else
-                await this._enterBrewing();
+                return;
+            }
+            if (this._stagingFailed)
+                this._stagingFailed = await this._recalculateStagingFailed();
+            await this._enterBrewing();
             return;
         }
 
         assert(this._stagedPosition);
 
         if (this._stagedPosition.merged()) {
             this._log("already merged into base some time ago");
+            // cleanup if this interrupted merge left a stale label behind
+            this._stagingFailed = false; // probably already false
```

Comment:
Just to avoid the implication that we are describing the `if` statement condition here:
```suggestion
            // cleanup in case this interrupted merge left a stale label behind
            this._stagingFailed = false; // probably already false
```

However, what if that label was not actually stale? We do accept that possibility in the _mergedSomeTimeAgo() case above... I suggest that we simplify and do nothing special for any merged() PRs. If they had the label and where somehow (manually?) merged despite it, keep the label. That would mean removing this comment/code _and_ the new comment in the _mergedSomeTimeAgo() case.